src/Main.hs

```diff
@@ -21,14 +26,17 @@ import           Language.Haskell.Stylish
 
 --------------------------------------------------------------------------------
 data StylishArgs = StylishArgs
-    { saVersion  :: Bool
-    , saConfig   :: Maybe FilePath
-    , saVerbose  :: Bool
-    , saDefaults :: Bool
-    , saInPlace  :: Bool
-    , saNoUtf8   :: Bool
-    , saFiles    :: [FilePath]
-    } deriving (Show)
+    { saVersion   :: Bool
+    , saConfig    :: Maybe FilePath
+    , saRecursive :: Maybe FilePath
```

Comment:
I think this should be a boolean, and if it is set, we should treat the files in `saFiles` as the paths to recursively search.  Optionally, we can rename `saFiles` to `saPaths`?

Reply:
Thanks @jaspervdj ! :) 

- _-r_ takes a path, in case it  was a folder a recursive search would be performed and _hs_ files would be resulted as its output. So, this is not just something to be switched on or off.

- _-e_ takes a path, which may be a folder or a  file; to define blacklist of files not to be formatted by stylish-haskell

The two switches are not necessary used together.

Using _-r_ or _-e_ switches would contribute to change the content of ```saFiles``` which would store file pathes no matter what. Certainly we can rename ```saFiles``` to ```saPaths```.